client/app/src/main/java/com/microsoft/research/karya/ui/registration/CreationCodeFragment.kt

```diff
@@ -1,33 +1,81 @@
-// Copyright (c) Microsoft Corporation.
-// Licensed under the MIT license.
-
 package com.microsoft.research.karya.ui.registration
 
 import android.content.Intent
 import android.os.Bundle
+import android.util.Log
 import android.util.TypedValue
+import androidx.fragment.app.Fragment
+import android.view.LayoutInflater
 import android.view.View
+import android.view.ViewGroup
+import androidx.navigation.findNavController
+import androidx.navigation.fragment.findNavController
 import com.microsoft.research.karya.R
+import com.microsoft.research.karya.data.service.KaryaAPIService
 import com.microsoft.research.karya.ui.base.BaseActivity
 import com.microsoft.research.karya.utils.SeparatorTextWatcher
-import kotlinx.android.synthetic.main.activity_creation_code.*
+import kotlinx.android.synthetic.main.fragment_creation_code.*
+import kotlinx.android.synthetic.main.fragment_creation_code.view.*
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.runBlocking
+
+/**
+ * A simple [Fragment] subclass.
+ * Use the [CreationCodeFragment.newInstance] factory method to
+ * create an instance of this fragment.
+ */
+
 
 private const val CREATION_CODE_LENGTH = 16
 
-class CreationCodeActivity : BaseActivity(useAssistant = true) {
+class CreationCodeFragment : Fragment() {
 
     /** Compute creation code text box length based on the creation code length */
     private val creationCodeEtMax = CREATION_CODE_LENGTH + (CREATION_CODE_LENGTH - 1) / 4
 
     /** Android strings */
-    private var accessCodePromptMessage: String = ""
-    private var invalidCreationCodeMessage: String = ""
-    private var creationCodeAlreadyUsedMessage: String = ""
+
+    private lateinit var registrationActivity: RegistrationActivity
+    private lateinit var baseActivity: BaseActivity
+    private lateinit var karyaAPI: KaryaAPIService
+
+    protected val ioScope = CoroutineScope(Dispatchers.IO)
+    protected val uiScope = CoroutineScope(Dispatchers.Main)
 
     override fun onCreate(savedInstanceState: Bundle?) {
-        setContentView(R.layout.activity_creation_code)
         super.onCreate(savedInstanceState)
+    }
+
+    override fun onCreateView(
+        inflater: LayoutInflater, container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View? {
+
+        registrationActivity = activity as RegistrationActivity
+        baseActivity = activity as BaseActivity
+        karyaAPI = baseActivity.karyaAPI
+
+        /** Initialising Strings  **/
+        // TODO: Remove this implementation when we fetch strings from resource
+        // This Initialisation is only needed to be done once for the first fragment
+        baseActivity.initialise()
```

Comment:
Why are we not fetching the strings from resources? We have already added those strings to the application.

Reply:
We are fetching it from resources. There is a different branch that converted every string fetch from DB to resources

Reply:
Yeah, saw that. Marking it as resolved.